runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java

```diff
@@ -550,6 +554,12 @@ public void process()
 
 	private void add(QueuedMessage message)
 	{
+		// Do not send message if the player is on tutorial island
+		if (TUTORIAL_ISLAND_REGIONS.contains(client.getLocalPlayer().getWorldLocation().getRegionID()))
```

Comment:
You should also null-check player, probably. As add can be called from login screen, I think, even though it does nothing.